src/fsharp/vs/service.fs

```diff
@@ -2289,10 +2297,7 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
     
 
     // STATIC ROOT: FSharpLanguageServiceTestable.FSharpChecker.parseFileInProjectCache. Most recently used cache for parsing files.
-    let parseFileInProjectCache = 
-        MruCache<ParseCacheLockToken, _, _>(parseFileInProjectCacheSize, 
-            areSame=AreSameForParsing3,
-            areSimilar=AreSubsumable3)
+    let parseFileInProjectCache = MruCache<_,_,_>(parseFileInProjectCacheSize, areSimilar = AreSimilarForParsing, areSame = AreSameForParsing)
```

Comment:
Please rename to ``parseFileCache``

Also please be explicit that this can be used from any caller thread, i.e. ``MruCache<AnyCallerThreadToken,_,_>`` like ``braceMatchCacheMru``

That said, I would feel a happier if we took a lock when accessing this cache as in the previous code - I don't think taking the lock on cache access will harm performance at all - and it might be better if the PR wasn't changing this cache from locked to lock free. I believe it's valid to do so (if not we have a bug in the braceMatchMruCache) - I'm just trying to minimize the size of the change in order to help you to get the tests to pass clean.